src/net_processing.cpp

```diff
@@ -3804,6 +3804,10 @@ bool PeerLogicValidation::ProcessMessages(CNode* pfrom, std::atomic<bool>& inter
     }
     CNetMessage& msg(msgs.front());
 
+    if (gArgs.IsArgSet("-logmessages")) {
+        LogMessage(*pfrom, msg.m_command, Span<const unsigned char>{(unsigned char*)msg.m_recv.data(), msg.m_recv.size()}, /* incoming */ true);
```

Comment:
Sorry that I didn't noticed on the earlier review today, but: if you rebase on master, this will unlock you the nice `MakeUCharSpan` (see commit e63dcc3a6752e7d406e7a650c2d6c2e95cd39aab, introduced with PR #19326 that was merged 3 days ago), not needing to cast anything anymore. :tada:

Reply:
Thanks, I'll do this!  Another upcoming improvement is #19660, but I'll try and leave that for a follow-up as not to increase the scope too far here.

Reply:
Turns out that wasn't optional.  Much nicer this way anyways :)

Reply:
Never mind